Add JUnit tests for ReviewedCommand, ShowCommand and PriorityCommand

g-lilian opened the pull request.

g-lilian committed:
Add JUnit tests for ReviewedCommand, ShowCommand and PriorityCommand

g-lilian committed:
Refactor to pass checkstyle

Zhilin-Huang commented on src/test/java/seedu/tp/commands/PriorityCommandTest.java:
```diff
@@ -0,0 +1,73 @@
+        expectedEventOutputDefault.append("Priority has been updated:" + System.lineSeparator());
+        expectedEventOutputDefault.append("Event 1 | New priority: Not indicated" + System.lineSeparator());
+
+        PriorityCommand priorityCommand = new PriorityCommand(FULL_FLASHCARD_LIST, 0, new Ui(), PriorityLevel.DEFAULT);
```
Use copy constructor for `FULL_FLASHCARD_LIST` to avoid affecting other tests

g-lilian replied:
I'm not sure why the build is still failing. I stopped using FULL_FLASHCARD_LIST in ReviewedCommand and PriorityCommand, and created a new list instead, so it shouldn't be modifying FULL_FLASHCARD_LIST.

Zhilin-Huang replied:
Hmmmm that's weird. Could you try running the tests in your local and see what are the expected and actual output?

g-lilian replied:
Thanks for all the help, the checks finally passed...

Zhilin-Huang commented on src/test/java/seedu/tp/commands/PriorityCommandTest.java:
```diff
@@ -0,0 +1,73 @@
+        expectedEventOutputHigh.append("Priority has been updated:" + System.lineSeparator());
+        expectedEventOutputHigh.append("Event 1 | New priority: ***" + System.lineSeparator());
+
+        PriorityCommand priorityCommand = new PriorityCommand(FULL_FLASHCARD_LIST, 0, new Ui(), PriorityLevel.HIGH);
```
Use copy constructor for `FULL_FLASHCARD_LIST` to avoid affecting other tests

Zhilin-Huang commented on src/test/java/seedu/tp/commands/PriorityCommandTest.java:
```diff
@@ -0,0 +1,73 @@
+        expectedPersonOutput.append("Priority has been updated:" + System.lineSeparator());
+        expectedPersonOutput.append("Person 1 | New priority: **" + System.lineSeparator());
+
+        PriorityCommand priorityCommand = new PriorityCommand(FULL_FLASHCARD_LIST, 1, new Ui(), PriorityLevel.MEDIUM);
```
Use copy constructor for `FULL_FLASHCARD_LIST` to avoid affecting other tests

Zhilin-Huang commented on src/test/java/seedu/tp/commands/PriorityCommandTest.java:
```diff
@@ -0,0 +1,73 @@
+        expectedOtherOutput.append("Priority has been updated:" + System.lineSeparator());
+        expectedOtherOutput.append("Title 1 | New priority: *" + System.lineSeparator());
+
+        PriorityCommand priorityCommand = new PriorityCommand(FULL_FLASHCARD_LIST, 2, new Ui(), PriorityLevel.LOW);
```
Use copy constructor for `FULL_FLASHCARD_LIST` to avoid affecting other tests

Zhilin-Huang commented on src/test/java/seedu/tp/commands/ReviewedCommandTest.java:
```diff
@@ -0,0 +1,39 @@
+        expectedOutput.append("You have marked the following flashcard as Reviewed:" + System.lineSeparator());
+        expectedOutput.append("Event 1" + System.lineSeparator());
+
+        ReviewedCommand reviewedCommand = new ReviewedCommand(FULL_FLASHCARD_LIST, 0, new Ui());
```
Use copy constructor for `FULL_FLASHCARD_LIST` to avoid affecting other tests

Zhilin-Huang reviewed (changes requested):
I suspect the reason why `ListCommandTest` is failing is that the `FULL_FLASHCARD_LIST` is altered by other tests

g-lilian committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T14-1/tp

g-lilian committed:
Fix assertion error for ListCommand

Instantiated new flashcardList in PriorityCommand to avoid changing FULL_FLASHCARD_LIST in ListCommand.

g-lilian committed:
Fix assertion error for ListCommand again

Created new flashcardList in ReviewedCommand as well, instead of using FULL_FLASHCARD_LIST.

g-lilian committed:
Try to fix assertion error in ListCommand

Replaced FULL_FLASHCARD_LIST constant with newly initialized list.

g-lilian committed:
Update ParserTest for Show, Reviewed, and Priority commands

g-lilian committed:
Fix to pass checkstyle

g-lilian committed:
Implement exception handling for invalid enum in PriorityCommand

Parser now throws an InvalidInputFormatException if the user enters any priority other than the 4 enum values (HIGH, MEDIUM, LOW, DEFAULT).

g-lilian committed:
Fix PriorityCommand to pass checkstyle

Minor fix to remove unused imports and changed thrown exception to be consistent with PriorityCommandTest.

g-lilian committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113-T14-1/tp

g-lilian committed:
Remove constants from Priority and ReviewedCommandTest

Removed constants in order to fix assertion error caused by constants being changed across different test classes.

g-lilian commented:
Will merge this long-overdue PR since all checks have finally passed.
